From d58a9c0a771114141c243ad6b08b99d5bb7a1b0c Mon Sep 17 00:00:00 2001
Message-Id: <d58a9c0a771114141c243ad6b08b99d5bb7a1b0c.1366916831.git.Jim.Somerville@windriver.com>
From: Ryan Finnie <ryan@finnie.org>
Date: Mon, 9 May 2011 17:38:40 -0700
Subject: [PATCH 1/1] Change linking order of -lpopt

Recent binutils apparently care about the linking order.  Reorder the
checkisomd5/implantisomd5 targets to put -lpopt at the end to avoid
compile failure. (Ubuntu Bug#771038)

Signed-off-by: Ryan Finnie <ryan@finnie.org>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 2ed24a5..8d61b4c 100644
--- a/Makefile
+++ b/Makefile
@@ -24,10 +24,10 @@ all: implantisomd5 checkisomd5 pyisomd5sum.so libimplantisomd5.a libcheckisomd5.
 	$(CC) -c -O $(CFLAGS) -o $@ $<
 
 implantisomd5: implantisomd5.o libimplantisomd5.a
-	$(CC) -lpopt $(CFLAGS) implantisomd5.o libimplantisomd5.a -o implantisomd5
+	$(CC) $(CFLAGS) implantisomd5.o libimplantisomd5.a -lpopt -o implantisomd5
 
 checkisomd5: checkisomd5.o libcheckisomd5.a
-	$(CC) -lpopt $(CFLAGS) checkisomd5.o libcheckisomd5.a -o checkisomd5
+	$(CC) $(CFLAGS) checkisomd5.o libcheckisomd5.a -lpopt -o checkisomd5
 
 libimplantisomd5.a: libimplantisomd5.a(libimplantisomd5.o md5.o)
 
-- 
1.7.4.1

